inspire_utils/logging.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+
+def log_error_with_extra(logger, message, *args, **kwargs):
+    """Log error with stack trace and locals information.
+
+    By default, enables stack trace information in logging messages, so that stacktrace and locals appear in Sentry.
+    """
+    kwargs.setdefault('extra', {}).setdefault('stack', True)
+    return logger.error(message, *args, **kwargs)
```

Comment:
What about using a decorator for the logger.error function instead?
That way you don't have to pass around the logger, that can be easily be 'fixed' with the functools.partial function though:

```python
log_error = partial(log_error_with_extra, logging.getLogger(__name__))
```

Reply:
I agree with @david-caro: it seems to me that the difficulty of remembering to call this method from `inspire-utils` seems comparable to remembering to add the `extra` kwarg, so this API doesn't seem super useful...